usr/src/psm/stand/boot/aarch64/common/arch_timer.c

```diff
@@ -63,6 +63,7 @@ arch_timer_mask_irq(void)
 	ctl = ops->read_cnt_ctl();
 	ctl |= CNT_CTL_IMASK;
 	ops->write_cnt_ctl(ctl);
+	isb();
```

Comment:
Would it make sense to put the `isb` in the write/read implementations, and not after each call to them?  That way they can't be forgotten, and we can leave a comment saying why more easily.

Reply:
I can do that, however if we do that then should we also apply this pattern to all of the other applicable registers in controlregs.h?

There are several other examples of controlreg calls immediately flowed by an isb() (write_s1e1r(), write_sctlr(), write_cpacr_el1(), write_ttbr0(),  write_tcr(), and write_vbar()).

The isb() preceding read() in this specific case is more to account for synchronization in tight loops (a plausible scenario for arch_timer_count()).

Reply:
It feels like if we need the barrier all the time, we should put the barrier in the function that does the read, yeah, but I'm ok with you not doing that right now for all the other registers, and we can do that separately.

Reply:
Sounds good. I'll update this later today with your recommended change.